core/src/main/scala/akka/persistence/cassandra/query/CassandraReadJournalConfig.scala

```diff
@@ -140,33 +147,33 @@ import scala.concurrent.duration._
     TimeBucket(date.toInstant(ZoneOffset.UTC).toEpochMilli, writePluginConfig.bucketSize)
   }
 
-  val deserializationParallelism: Int = config.getInt("deserialization-parallelism")
+  val deserializationParallelism: Int = readConfig.getInt("deserialization-parallelism")
 
-  val pluginDispatcher: String = config.getString("plugin-dispatcher")
+  val pluginDispatcher: String = readConfig.getString("plugin-dispatcher")
 
   val keyspace: String = writePluginConfig.keyspace
   val targetPartitionSize: Long = writePluginConfig.targetPartitionSize
   val table: String = writePluginConfig.table
   val pubsubNotification: Duration =
     writePluginConfig.tagWriterSettings.pubsubNotification
   val eventsByPersistenceIdEventTimeout: FiniteDuration =
-    config.getDuration("events-by-persistence-id-gap-timeout", MILLISECONDS).millis
+    readConfig.getDuration("events-by-persistence-id-gap-timeout", MILLISECONDS).millis
 
   val eventsByTagGapTimeout: FiniteDuration =
-    config.getDuration("events-by-tag.gap-timeout", MILLISECONDS).millis
+    eventsByTagConfig.getDuration("gap-timeout", MILLISECONDS).millis
```

Comment:
I think events by tag needs its own settings class (for a different PR)

Reply:
yes, I have a few other things I would like to change in those classes so let me do that in a separate PR